### PR TITLE
Changed Authentication and User Redirection Logic and added Profile Pages

### PR DESCRIPTION
**Pull request by Domm6**
1. Refinement of User Role-Based Redirection: Users are now directed to dedicated pages tailored to their roles as mentors or mentees upon signup. This ensures a more personalized and relevant user experience right from the start.
2. Transition from Firebase to Session-Based Authentication: shifted from using Firebase for authentication to a session-based approach. This change enhances the integration of our authentication system with other backend services, providing a more cohesive and secure user experience and works with postgresSQL database.
3. Consistent Redirection Logic Across Login and Signup: The redirection logic that was applied to the signup process has now been mirrored on the login page. This consistency ensures that users are always directed to the appropriate page based on their roles, regardless of whether they are logging in or signing up for the first time.

Failed Login Page:
Screenshot 2024-06-26 at 11 45 47 AM

Example of Mentor Profile Page:
Screenshot 2024-06-26 at 11 46 01 AM

### CODE REVIEW

File: Backend/server.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      expires: new Date(Date.now() + (365 * 24 * 60 * 60 * 1000)) // 1 year in milliseconds
+    }
+  })
+);
+sessionStore.sync();
+
+app.use(userRoutes);
+
+// Route to get all posts, with associated users
+app.get('/posts', async (req, res) => {
```
Comment (jzwang43) [inline Backend/server.js]: Do we have the feature for posts? If not we can remove them?

File: Frontend/src/components/MentorDashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+function MentorDashboard() {
+    return(
+        <div>
+            <h1>Mentor Dashboard</h1>
+        </div>
+    )
+}
+
+export default MentorDashboard
```
Comment (jzwang43) [inline Frontend/src/components/MentorDashboard.jsx:35]: might be able to just combine the import into `export default function MentorDashboard() {...}` ?

File: Backend/Routes/users.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        [Op.or]: [{ email }]
+      }
+    });
+
+    if (existingUser) {
+      return res.status(400).json({ error: 'Email already exists' });
+    }
+
+    // Encrypt the password
+    const hashedPassword = await bcrypt.hash(password, 10);
```
Comment (wspalding) [inline Backend/Routes/users.js]: lets make 10 a constant value, that way it has a name, and if it needs to be referenced anywhere else, there is a single source of truth. 
In general any kind of "special number" should be a const

File: Frontend/src/components/SignUpPage.jsx [hunk 4]
```diff
@@ -1,43 +1,122 @@
...
+        },
+        body: JSON.stringify({ name, email, password, userRole }),
+        credentials: 'include'
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        const loggedInUser = data.user;
+
+        console.log('Signup successful');
```
Comment (jzwang43) [inline Frontend/src/components/SignUpPage.jsx]: For the production code we don't want console.log in it, you can either remove it or add a todo to clean up later

File: Backend/Routes/users.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    // Create a new user
+    const newUser = await User.create({
+      name,
+      email,
+      password: hashedPassword,
+      userRole
+    });
+
+    // Set the user in the session
```
Comment (jzwang43) [inline Backend/Routes/users.js:39]: I like the use of comments across the files, please keep this up so the code is easier to understand

File: Backend/Routes/signup.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const express = require('express')
+const router = express.Router()
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+const cors = require('cors');
+router.use(cors());
+
+router.get("/test", async (req, res) => {
```
Comment (sbrina813) [inline Backend/Routes/signup.js]: Is this in use? if it is just test use, shall we remove it from production code?

File: Backend/Routes/users.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+// Route for user login
+router.post('/users/login', async (req, res) => {
+  const { email, password } = req.body;
+
+  try {
+    // Find the user by email
+    const user = await User.findOne({ where: { email } });
+
+    if (!user) {
+      return res.status(401).json({ error: 'Invalid email or password' });
```
Comment (sbrina813) [inline Backend/Routes/users.js]: error: 'Invalid email' ? and ln66 for error: 'Invalid password' ?

File: Backend/server.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+      return res.status(401).json({ error: 'Unauthorized' });
+    }
+
+    // Retrieve the current user from the session
+    const currentUser = req.session.user;
+
+    // Create the post with the current user ID
+    const post = await Post.create({
+      ...req.body,
+      userId: currentUser.id
```
Comment (sbrina813) [inline Backend/server.js]: is userId the same as user.id ? why don't we just use id?

File: Backend/Routes/users.js (hunk 3 above)
Comment (Domm6) [reply Backend/Routes/users.js]: Resolved

File: Backend/server.js (hunk 1 above)
Comment (Domm6) [reply Backend/server.js]: Resolved

File: Frontend/src/components/MentorDashboard.jsx (hunk 2 above)
Comment (Domm6) [reply Frontend/src/components/MentorDashboard.jsx:35]: Resolved

File: Frontend/src/components/SignUpPage.jsx (hunk 4 above)
Comment (Domm6) [reply Frontend/src/components/SignUpPage.jsx]: Resolved

File: Backend/Routes/signup.js (hunk 6 above)
Comment (Domm6) [reply Backend/Routes/signup.js]: It was for test use, thanks for pointing out! I have removed it.

File: Backend/Routes/users.js (hunk 7 above)
Comment (Domm6) [reply Backend/Routes/users.js]: Resolved. I have updated it, one for email, one for password.

File: Backend/server.js (hunk 8 above)
Comment (Domm6) [reply Backend/server.js]: This was template code. I have removed it.

**Review by jzwang43 — APPROVED** [review]

**Pull request merged**